generator/internal/language/templates/rust/common/enum.mustache

```diff
@@ -18,13 +18,12 @@ limitations under the License.
 {{{.}}}
 {{/Codec.DocLines}}
 #[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
-pub struct {{Codec.Name}}(std::string::String);
+pub struct {{Codec.Name}}(std::borrow::Cow<'static, str>);
 
 impl {{Codec.Name}} {
-    /// Sets the enum value.
-    pub fn set_value<T: std::convert::Into<std::string::String>>(mut self, v: T) -> Self {
-        self.0 = v.into();
-        self
+    /// Creates a new {{Codec.Name}} instance.
+    pub const fn new(v: &'static str) -> Self {
+        Self(std::borrow::Cow::Borrowed(v))
```

Comment:
We also need a way to construct these from a `std::string::String`, in case a customer built a string dynamically, something like:

```suggestion
    /// Creates a new {{Codec.Name}} instance.
    pub const fn new(v: &'static str) -> Self {
        Self(std::borrow::Cow::Borrowed(v))
   }
   ... the existing code ...
}

impl std::convert::From<std::string::String> for {{Codec.Name}} {
  fn from(value: std::string::String) -> Self {
    Self(std::borrow::Cow::Owned(value))
  }
}
```

Reply:
Ah missed this, thank you. Added unit testing to verify this for a generated enum under `src/wkt/generated/` but open to suggestions on a better location.